core-bundle/src/Resources/contao/dca/tl_member.php

```diff
@@ -584,158 +580,4 @@ public function storeDateAdded($dc)
 		$this->Database->prepare("UPDATE tl_member SET dateAdded=? WHERE id=?")
 					   ->execute($time, $dc->id);
 	}
-
-	/**
-	 * Return the "toggle visibility" button
-	 *
-	 * @param array  $row
-	 * @param string $href
-	 * @param string $label
-	 * @param string $title
-	 * @param string $icon
-	 * @param string $attributes
-	 *
-	 * @return string
-	 */
-	public function toggleIcon($row, $href, $label, $title, $icon, $attributes)
-	{
-		if (Input::get('tid'))
-		{
-			$this->toggleVisibility(Input::get('tid'), (Input::get('state') == 1), (func_num_args() <= 12 ? null : func_get_arg(12)));
-			$this->redirect($this->getReferer());
-		}
-
-		// Check permissions AFTER checking the tid, so hacking attempts are logged
-		if (!System::getContainer()->get('security.helper')->isGranted(ContaoCorePermissions::USER_CAN_EDIT_FIELD_OF_TABLE, 'tl_member::disable'))
```

Comment:
You have kept this check in the `tl_content::toggleIcon()` method. Why did you remove it here?

Reply:
This security check (basic field access) is done by the generic implementation as well, since it's simple. `tl_content` needs to additionally check for access to the content element though (allow the operation), so it needs a button callback. Access to `act=toggle` (through URL manipulation) is then again done by `checkPermission` callback.

Reply:
Does this mean that we can remove this check?

https://github.com/contao/contao/blob/b5ffcb1e816ce920c5b22966985fdc45cc638c1a/core-bundle/src/Resources/contao/dca/tl_content.php#L1983-L1986

Because it is already checked in the generic method:

https://github.com/contao/contao/blob/b5ffcb1e816ce920c5b22966985fdc45cc638c1a/core-bundle/src/Resources/contao/drivers/DC_Table.php#L2703-L2707

Reply:
I see. 👍